brian2/stateupdaters/base.py

```diff
@@ -100,6 +103,16 @@ def apply_stateupdater(equations, variables, method, group_name=None):
         abstract_code : str
             The code integrating the given equations.
         '''
+        if isinstance(method, basestring):
+            method_key = method
+        elif hasattr(method, '__call__'):
+            method_key = method.__class__.__name__
+        else:
+            method_key = tuple(method)
+        cache_key = (equations, _hashable(variables), method_key)
+        if cache_key in StateUpdateMethod._state_update_cache:
+            return StateUpdateMethod._state_update_cache[cache_key]
+
```

Comment:
This looks like another case where wrapping ``_init__`` or the class could work to automate this.